spring-mockito/src/test/java/com/baeldung/reader/FlowerReaderServiceUnitTest.java

```diff
@@ -0,0 +1,51 @@
+package com.baeldung.reader;
+
+import com.baeldung.app.api.Flower;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
+public class FlowerReaderServiceUnitTest {
+
+    @MockBean
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private FlowerReaderService flowerReader;
+
+    @Test
+    public void whenCallingWithJsonAsString_thenReturnsExpectedFlowerObject() throws JsonProcessingException {
+
+        String flowerAsJson = "{\"name\": \"rose\", \"petals\": 100}";
+        String roseName = "rose";
+        Integer rosePetals = 100;
+        Flower rose = new Flower(roseName, rosePetals);
+
+        Mockito.when(objectMapper.readValue(flowerAsJson, Flower.class)).thenReturn(rose);
+        Flower expectedRose = flowerReader.readFlower(flowerAsJson);
```

Comment:
It is not necessary to provide a specific string to specify the Mock behavior. You can simply use any(String.class) as the parameter to readValue instead of an actual string. Then you can specify any string you want when you actually make the call to your service's readFlower method.